### DISCUSSION TITLE
Facing "R3InjectorError(Standalone[StorybookWrapperComponent])" error

### DISCUSSION THREAD

**Discussion by hagealex** [storybookjs/storybook#25711] [Help]
### Summary

Hello everyone!

I'm working with Storybook `7.6.10` and Angular `17.1.0`. I have one component `InputTextComponent` that renders an input field. Nothing special so far. The template for this component renders another component as a wrapper for the input. This component is responsible to render the associated label and validates the input:
```
<ui-form-control
   ...som props here...
>
    <input
       ...some props here...
    />
</ui-form-control>
```

So, also nothing special. But now my issue: The `FormControlComponent` ("`<ui-form-control>`") makes use of a custom pipe:

```
constructor(
        private rootFormGroup: FormGroupDirective,
        private textResourcePipe: TextResourcePipe
    ) {}
```

My story for this text input component looks like this:
```
const meta: Meta<InputTextComponent> = {
    title: 'InputFields/Text',
    decorators: [
        moduleMetadata({
          declarations: [FormControlComponent, TextResourcePipe],
          imports: [ReactiveFormsModule]
        }),
        applicationConfig({
            providers: [FormGroupDirective],
        }),
        componentWrapperDecorator(
          (story) => `<form [formGroup]="form">${story}</form>`
        ),
    ],
    component: InputTextComponent,
    tags: ['autodocs'],
    render: (args: InputTextComponent) => ({
        props: {
            ...args,
          controlName: 'text',
          form: formGroup,
        },
    }),
    argTypes: {},
}
```

When rendering this story I get the following error:
`Error: R3InjectorError(Standalone[StorybookWrapperComponent])[TextResourcePipe -> TextResourcePipe -> TextResourcePipe]: `

When I remove the `TextResourcePipe` from the stories declarations I get this error:
`Error: Type FormControlComponent is part of the declarations of 2 modules: StorybookComponentModule and StorybookComponentModule! Please consider moving FormControlComponent to a higher module that imports StorybookComponentModule and StorybookComponentModule. You can also create a new NgModule that exports and includes FormControlComponent then import that NgModule in StorybookComponentModule and StorybookComponentModule`

After some hours of research I still haven't found out what I'm doing wrong here.

Appreciate any help! :)

Kind regards

**Comment by vvs09** [comment]
Import the forms control component module under imports.if textResource pipe is part of the formaControlModule you can exclude the declarations.

moduleMetadata({
 imports:[ReactiveFormsModule, FormControlModule]
        }),

**Reply by hagealex** [reply]
Hello @vvs09 Thanks for your support here! I tried as you suggested and then I get this error: `Error: Unexpected directive 'FormControlComponent' imported by the module 'StorybookComponentModule'. Please add an @NgModule annotation.`

**Comment by Marklb** [comment]
I haven't had a chance to figure out how to reproduce the error, from the snippets you provided, but I tried to quickly make a reproduction and didn't get any problem. I don't know if I made wrong assumptions about parts of the code that you didn't provide or if something has been fixed in the alpha.

I may try to reproduce the error again, but if you could provide a reproduction that allows me to not make assumptions about `TextResourcePipe`, `InputTextComponent`, `FormControlComponent` and `FormControlModule` then that probably would help find the problem faster.

Also, if the error about being declared by 2 modules is what I am thinking then it may be fixed in `8.0.0-alpha.11`

**Reply by hagealex** [reply]
@Marklb I will check if I still get the error in `8.0.0-alpha.11` and then come back.

**Reply by hagealex** [reply]
@Marklb Setting up the 8.0 alpha version causes other issues. So I'll try to stick with the current version to fix it there 😄

**Comment by hagealex** [comment]
I solved my issue. The problem was that I didn't add the Pipe to the `providers` array of `applicationConfig`. Now it's working as expected.
